assets/scripts/submit.js

```diff
@@ -37,137 +42,89 @@
-          submitButton.classList.add('c-btn--highlight')
-          if (localStorage && typeof body.two_factor_trusted_device_token == 'string') {
-            localStorage.setItem(twoFactorTrustedDeviceTokenKey, body.two_factor_trusted_device_token)
-          }
-          if (body.redirect) {
-            window.location = body.redirect
-          } else {
-            form.submit()
-          }
-        } else {
-          showError(body.error)
-          twoFactorPasscodeInput.classList.add('is-error')
-          twoFactorPasscodeInput.select()
-        }
-      }).catch(showError)
-    }).catch(showError)
+    })
+      .then(function(response) {
+        const loginSuccess = response.status < 400
+        response
+          .json()
+          .then(function(body) {
+            if (loginSuccess) {
+              submitButton.childNodes[1].innerHTML =
+                '<svg width="16" height="16"><use xlink:href="#fa-check"/></svg>'
+              submitButton.classList.add('c-btn--highlight')
+              if (body.redirect) {
+                window.location = body.redirect
+              }
+            } else {
+              showError(body.error)
+              passphraseInput.classList.add('is-error')
+              passphraseInput.select()
+            }
+          })
+          .catch(showError)
+      })
+      .catch(showError)
   }
-
-  function renderTwoFactorForm(twoFactorToken) {
-    for (let i = 0; i < twoFactorForms.length; i++) {
-      twoFactorForms[i].classList.remove('u-hide')
-    }
-    for (let i = 0; i < passwordForms.length; i++) {
-      passwordForms[i].classList.add('u-hide')
-    }
-    if (errorPanel) errorPanel.textContent = ''
-    submitButton.removeAttribute('disabled')
-    twoFactorTokenInput.value = twoFactorToken
-    twoFactorPasscodeInput.value = ''
-    twoFactorPasscodeInput.focus()
-    loginForm.removeEventListener('submit', onSubmitPassphrase)
-    loginForm.addEventListener('submit', onSubmitTwoFactorCode)
-  }
-
   loginForm && loginForm.addEventListener('submit', onSubmitPassphrase)
 
-  resetForm && resetForm.addEventListener('submit', function(event) {
-    event.preventDefault()
-    const label = window.password.getStrength(passphraseInput.value).label
-    if (label == 'weak') {
-      return false
-    } else {
-      resetForm.submit()
-    }
-  })
-
-  resetForm && passphraseInput.addEventListener('input', function(event) {
-    const label = window.password.getStrength(event.target.value).label
-    submitButton[label == 'weak' ? 'setAttribute' : 'removeAttribute']('disabled', '')
-  })
+  // Used for passphrase reset
+  resetForm &&
+    resetForm.addEventListener('submit', function(event) {
+      event.preventDefault()
+      const label = window.password.getStrength(passphraseInput.value).label
+      if (label == 'weak') {
+        return false
+      } else {
+        resetForm.submit()
+      }
+    })
+
+  resetForm &&
```

Comment:
s/resetForm/passphraseInput/ ?

Reply:
IIRC, this event listener is only used for reset passphrase page, I don't think we need it in the long form

(https://github.com/cozy/cozy-stack/blob/master/assets/scripts/submit.js#L167)